lib/manageiq/release/pull_request_blaster_outer.rb

```diff
@@ -0,0 +1,136 @@
+        puts "+++ blasting #{repo.github_repo}..."
+
+        repo.git
+        repo.fetch(output: false)
+        repo.checkout(pr_branch, "origin/#{branch}")
+
+        run_script
+
+        result = false
+        if !commit_changes
+          puts "!!! Failed to commit changes. Perhaps the script is wrong or #{repo.github_repo} is already updated."
+        elsif dry_run
+          result = "Committed but is dry run"
+        else
+          puts "Do you want to open a pull request on #{repo.github_repo} with the above changes? (Y/N)"
+          answer = $stdin.gets.chomp
+          if answer.upcase.start_with?("Y")
+            fork_repo unless forked?
+            push_branch
+            result = open_pull_request
+          end
+        end
+        puts "--- blasting #{repo.github_repo} complete"
+        result
+      end
+
+      private
+
+      def github
+        ManageIQ::Release.github
+      end
+
+      def forked?
+        github.repos(github.login).any? { |m| m.name == repo.name }
+      end
+
+      def with_status
+        calling_method = caller_locations(1,1)[0].label
+        puts "+++ #{calling_method}..."
+        result = yield
+        puts "--- #{calling_method}: #{result}"
+        result
+      end
+
+      def fork_repo
+        with_status do
+          github.fork(repo.github_repo)
+          until forked?
+            print "."
+            sleep 3
+          end
+        end
+      end
+
+      def run_script
+        with_status do
+          Dir.chdir(repo.path) do
+            `#{script}`
+          end
+        end
+      end
+
+      def commit_changes
+        with_status do
+          Dir.chdir(repo.path) do
+            begin
+              repo.git.add("-v", ".")
+              repo.git.commit("-m", message)
+              repo.git.show
+              if dry_run
+                puts "!!! --dry-run enabled: If the above commit in #{repo.path} looks good, run again without dry run to fork the repo, push the branch and open a pull request."
+              end
+              true
+            rescue MiniGit::GitError => e
+              e.status.exitstatus == 0
+            end
+          end
+        end
+      end
+
+      def origin_remote
+        "pr_blaster_outer"
+      end
+
+      def origin_url
+        "git@github.com:#{github.login}/#{repo.name}.git"
+      end
+
+      def pr_branch
+        "pr_blaster_outer_#{branch}_#{File.basename(script)[0,3]}"
+      end
+
+      def pr_base
+        "#{github.login}:#{pr_branch}"
+      end
+
+      def push_branch
+        with_status do
+          Dir.chdir(repo.path) do
+            repo.git.remote("add", origin_remote, origin_url) unless repo.remote?(origin_remote)
+            repo.git.push("-f", origin_remote, "#{pr_branch}:#{pr_branch}")
+          end
+        end
+      end
+
+      def open_pull_request
+        with_status do
+          pr = github.create_pull_request(repo.github_repo, branch, pr_base, message[0,72], message[0,72])
```

Comment:
Note, this blows up if for example, you've already opened a PR for this script before... (your fork has the `pr_branch` already).... This could be better detected to handle re-pushing if we want